### PR TITLE
Don't unconditionally return false when checking for upgradability

### LINKED ISSUE #144: brew bundle doesn't update outdated packages
I have this `rbenv-aliases` version `1.0.1` installed

```shell
$ brew info rbenv-aliases
rbenv-aliases: stable 1.1.0, HEAD
Make aliases for Ruby versions
https://github.com/tpope/rbenv-aliases
/usr/local/Cellar/rbenv-aliases/1.0.1 (7 files, 6.5K)
  Built from source
From: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/rbenv-aliases.rb
```

If I try to list the outdated packages, the `rbenv-aliases` package is there

```shell
$ brew outdated
...
rbenv-aliases (1.0.1 < 1.1.0)
...
```

But if I run `brew bundle --file=Brewfile` againt the following `Brewfile`, nothing gets updated.

```shell
$ cat Brewfile
brew 'rbenv'
brew 'rbenv-vars'
brew 'rbenv-aliases'
brew 'ruby-build'

$ brew bundle --file=Brewfile
Succeeded in installing rbenv
Succeeded in installing rbenv-vars
Succeeded in installing rbenv-aliases
Succeeded in installing ruby-build
```

After that I still have `rbenv-aliases` version `1.0.1`.
any idea?

### PR DESCRIPTION
**Pull request by ilovezfs**
Closes #144

### CODE REVIEW

**Comment by mikemcquaid** [comment]
@ilovezfs Could you consider adding a failing test that this fixes? I'm not sure I understand the fix.

File: lib/bundle/brew_installer.rb [hunk 1]
```diff
@@ -55,7 +55,8 @@ def self.installed_formulae
     end
 
     def self.upgradable_formulae
-      outdated_formulae - pinned_formulae
+      upgradable = outdated_formulae - pinned_formulae
+      upgradable.nil? ? nil : upgradable.map { |u| u[:name] }
```
Comment (mikemcquaid) [inline lib/bundle/brew_installer.rb]: It seems to me this method should always return an array.

**Comment by ilovezfs** [comment]
> Could you consider adding a failing test that this fixes? I'm not sure I understand the fix.
@mikemcquaid This is the minimal change to fix it, but you may actually want more extensive changes.

The problem is that installed_formulae is an array of formula names, whereas outdated_formulae and pinned_formulae are arrays of formulae. The output of upgradable_formulae is treated as if it's the same as installed_formulae, an array of formula names, but in reality it was the difference between outdated_formulae and pinned_formulae, both of which are arrays of formulae not formula names.

**Comment by mikemcquaid** [comment]
> The problem is that installed_formulae is an array of formula names, whereas outdated_formulae and pinned_formulae are arrays of formulae. The output of upgradable_formulae is treated as if it's the same as installed_formulae, an array of formula names, but in reality it was the difference between outdated_formulae and pinned_formulae, both of which are arrays of formulae not formula names.

Thanks for explaining!

File: lib/bundle/brew_installer.rb [hunk 2]
```diff
@@ -55,7 +55,8 @@ def self.installed_formulae
     end
 
     def self.upgradable_formulae
-      outdated_formulae - pinned_formulae
+      upgradable = outdated_formulae - pinned_formulae
+      upgradable.nil? ? [] : upgradable.map { |u| u[:name] }
```
Comment (mikemcquaid) [inline lib/bundle/brew_installer.rb]: `(outdated_formulae - pinned_formulae).map { |f| f[:name] }` would work?

Comment (ilovezfs) [inline lib/bundle/brew_installer.rb]: Yes. Or we could return names from outdated_formulae and pinned_formulae.

Comment (mikemcquaid) [inline lib/bundle/brew_installer.rb]: Actually, yeh, that seems like the best fix :+1:

**Comment by ilovezfs** [comment]
@mikemcquaid How's that, with compact?

File: lib/bundle/brew_installer.rb [hunk 3]
```diff
@@ -59,11 +59,11 @@ def self.upgradable_formulae
     end
 
     def self.outdated_formulae
-      @outdated_formulae ||= Bundle::BrewDumper.formulae.select { |f| f[:outdated?] }
+      @outdated_formulae ||= Bundle::BrewDumper.formulae.map { |f| f[:name] if f[:outdated?] }.compact
```
Comment (xu-cheng) [inline lib/bundle/brew_installer.rb]: `compact` is unnecessary as there won't be `nil` inside the input list.

**Comment by xu-cheng** [comment]
You will need to update corresponding tests https://github.com/Homebrew/homebrew-bundle/blob/master/spec/brew_installer_spec.rb#L36-L58

File: lib/bundle/brew_installer.rb (hunk 3 above)
Comment (ilovezfs) [inline lib/bundle/brew_installer.rb]: Um,
```
irb(main):001:0> [ 3, 4, 5, 6 ].map { |n| (n+3) if n > 3 }
=> [nil, 7, 8, 9]
```

Comment (xu-cheng) [inline lib/bundle/brew_installer.rb]: Oh, sorry my fault. I missed the `select` here.

Comment (ilovezfs) [inline lib/bundle/brew_installer.rb]: No problem. This was in lieu of doing .select {g(x)}.map {f(x)}. Is there any even better way than .map {f(x) if g(x)}.compact?

**Comment by ilovezfs** [comment]
@xu-cheng better?

**Comment by xu-cheng** [comment]
:+1: LGTM. But one last nit: could you update the commit message? Thanks.

**Comment by ilovezfs** [comment]
@xu-cheng commit message is now not `nil`.

**Comment by xu-cheng** [comment]
Thanks @ilovezfs

**Comment by ilovezfs** [comment]
@xu-cheng You're welcome :)

**Pull request merged**